apps/valorantstats/valorant_stats.star

```diff
@@ -0,0 +1,186 @@
+    #turn riot_nameTag into Name and Tag
+    #riot_nameTag = config.str("riot_nameTag")
+    #regex concat string before # and after # into Name and Tag var
+    # If no riot id provided, return rank 1 stats
+    if riot_name == "":
+        print("riot ID not set, checking lb for spot 1")
+        lb_data1, lb_data3 = checkRank1(lb_data, provided_val_tag_name)
+        provided_val_tag_name = lb_data3
+        print("val_tag_api_link is: " + val_tag_api_link)
+        val_tag_api_link = "https://api.henrikdev.xyz/valorant/v2/mmr/na/" + lb_data1
+        print("val_tag_api_link is: " + val_tag_api_link)
+        print("checked lb, api link set as: " + val_tag_api_link)
+
+    # call get api data function (check if have cache or make fresh api call) w/ ttl of 300 to not rate limit api
+    user_stat_data = getAPIDataCacheOrHTTP(val_tag_api_link, data, ttl_seconds = 300)
+
+    # encode json data for parsing later
+    user_stat_data = json.encode(user_stat_data)
+
+    #call user stats function passing in API JSON data and provided_val_tag_name
+    if user_stat_data == None:
+        print("yo u aint got NO data my boy das not good")
+        # draft fail render screen here
+        # it wont reach this point as if it has no data it will either 404/error from api HTTP req OR it will default to #1 leaderboard spot
+
+    else:
+        print("You have data, proceeding")
+
+        # set data variables from cached/parsed json
+        rank = json.decode(user_stat_data)["data"]["current_data"]["currenttierpatched"]
+        json_image = json.decode(user_stat_data)["data"]["current_data"]["images"]["small"]
+
+        # exception case if rank is set to none set text/image to unranked so it can still pass data
+        if rank == None:
+            rank = "Unranked"
+            json_image = "https://trackercdn.com/cdn/tracker.gg/valorant/icons/tiersv2/0.png"
+
+        # get base64 image from json png url
+        rank_image = getImage(json_image)
+
+    # render data
+    print(provided_val_tag_name + ", " + rank)
+    return renderStats(rank, rank_image, provided_val_tag_name)
+
+# caching!
+def getAPIDataCacheOrHTTP(val_tag_api_link, data, ttl_seconds):
+    user_cache = cache.get(data)
+    if user_cache != None:
+        print("Displaying cached user data")
+        rep = json.decode(user_cache)
+        data = rep.json()
+        print(data)
+    else:
+        print("No cached data. Hitting API")
+        rep = http.get(val_tag_api_link)
+        if rep.status_code != 200:
+            fail("VALORANT API request failed with status %d", rep.status_code)
+        cache.set(val_tag_api_link, json.encode(rep.json()), ttl_seconds = ttl_seconds)
+        data = rep.json()
+        print("Hit API and cached data")
+    return data
+
+# get image from predefined json image url from stats api
+def getImage(url):
+    if url:
+        print("Getting image from: " + url)
+        response = http.get(url)
```

Comment:
Would it make sense to add some caching around this call? Otherwise we'll be fetching the image every single time a Tidbyt is trying to display this app.

Reply:
Good call, that would definitely make sense. I'll implement it this weekend.

Reply:
I've added some caching here